specification/batch/Batch/routes.tsp

```diff
@@ -0,0 +1,2755 @@
+  )
+  @example(
+    "./examples/PoolAdd_MountConfiguration.json",
+    "Add a pool with mount drive specified"
+  )
+  @route("/pools")
+  Add is RpcOperation<
+    BatchClientRequestHeaders & {
+      @doc("Type of content")
+      @header("content-type")
+      contentType: "application/json; odata=minimalmetadata";
+
+      @body
+      @doc("The Pool to be added.")
+      pool: BatchPool;
+    },
+    BatchResponseHeaders & {
+      @doc("A process exit code.")
+      @statusCode
+      code: "201";
+
+      @doc("The OData ID of the resource to which the request applied")
+      @header("DataServiceId")
+      DataServiceId: string;
+    },
+    {},
+    BatchError
+  >;
+
+  @summary("Lists all of the Pools in the specified Account.")
+  @doc("Lists all of the Pools in the specified Account.")
+  @route("/pools")
+  @example("./examples/PoolList_Basic.json", "Pool list")
+  @get
+  List is RpcOperation<
+    BatchApplicationListHeaders & {
+      @doc("""
+An OData $filter clause. For more information on constructing this filter, see
+https://docs.microsoft.com/en-us/rest/api/batchservice/odata-filters-in-batch#list-pools.
+""")
+      @query
+      $filter?: string;
+
+      @doc("An OData $select clause.")
+      @query
+      $select?: string;
+
+      @doc("An OData $expand clause.")
+      @query
+      $expand?: string;
+    },
+    BatchResponseHeaders & BatchPoolListResult,
+    {},
+    BatchError
+  >;
+
+  @summary("Deletes a Pool from the specified Account.")
+  @doc("""
+When you request that a Pool be deleted, the following actions occur: the Pool
+state is set to deleting; any ongoing resize operation on the Pool are stopped;
+the Batch service starts resizing the Pool to zero Compute Nodes; any Tasks
+running on existing Compute Nodes are terminated and requeued (as if a resize
+Pool operation had been requested with the default requeue option); finally,
+the Pool is removed from the system. Because running Tasks are requeued, the
+user can rerun these Tasks by updating their Job to target a different Pool.
+The Tasks can then run on the new Pool. If you want to override the requeue
+behavior, then you should call resize Pool explicitly to shrink the Pool to
+zero size before deleting the Pool. If you call an Update, Patch or Delete API
+on a Pool in the deleting state, it will fail with HTTP status code 409 with
+error code PoolBeingDeleted.
+""")
+  @example("./examples/PoolDelete.json", "Pool delete")
+  @route("/pools/{poolId}")
+  @delete
+  Delete is RpcOperation<
+    BatchClientRequestHeaders & BatchPoolHeaders,
+    DeleteResponseHeaders,
+    {},
+    BatchError
+  >;
+
+  @doc("Gets basic properties of a Pool.")
+  @example("./examples/PoolExists.json", "Check Pool Exists")
+  @route("/pools/{poolId}")
+  @head
+  Exists is RpcOperation<
+    BatchClientRequestHeaders & BatchPoolHeaders,
+    PoolDoesntExistResponseHeaders | PoolDoesExistResponseHeaders,
```

Comment:
I meet the following error with latest tsp compilor (0.45.2):
![image](https://github.com/Azure/azure-rest-api-specs/assets/70930885/80338168-58bf-43b0-8a30-16225ff723a2)
It seems illegal when anonymous model is assignable to type 'Model'

Reply:
comment from my colleague:
![image](https://github.com/Azure/azure-rest-api-specs/assets/70930885/c466a37b-d3c0-4ce7-a1f1-4634de9e2ada)